sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbird/integration/service/AuthenticationHelperService.java

```diff
@@ -0,0 +1,154 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.mosip.esignet.api.dto.*;
+import io.mosip.esignet.api.exception.KycAuthException;
+import io.mosip.esignet.api.exception.SendOtpException;
+import io.mosip.esignet.api.util.ErrorConstants;
+import io.mosip.esignet.sunbird.integration.dto.InsurenceResponceDto;
+import io.mosip.esignet.sunbird.integration.dto.KycAuthRequestDto;
+import io.mosip.esignet.sunbird.integration.dto.SearchRequestDto;
+import io.mosip.kernel.core.http.ResponseWrapper;
+import io.mosip.kernel.signature.dto.JWTSignatureRequestDto;
+import io.mosip.kernel.signature.dto.JWTSignatureResponseDto;
+import io.mosip.kernel.signature.service.SignatureService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.*;
+
+@Component
+@Slf4j
+public class AuthenticationHelperService {
+
+    private final String Full_Name="fullName";
+    private final String DOB="dob";
+    private final String Policy_Number="policyNumber";
+    public static final String OIDC_PARTNER_APP_ID = "OIDC_PARTNER";
+    private static final Base64.Encoder urlSafeEncoder = Base64.getUrlEncoder().withoutPadding();
+    public static final String CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    @Value("${mosip.esignet.mock.authenticator.send-otp}")
+    private String sendOtpUrl;
+    @Value("${mosip.esignet.mock.authenticator.kyc-auth-url}")
+    private String kycAuthUrl;
+    @Value("${mosip.esignet.mock.authenticator.ida.otp-channels}")
+    private List<String> otpChannels;
+
+    @Value("${mosip.esignet.authenticator.default.auth-factor.kba.field.key:eq}")
+    private String fieldKey;
+
+    @Autowired
+    private SignatureService signatureService;
+    @Autowired
+    private RestTemplate restTemplate;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+
+    protected static LocalDateTime getUTCDateTime() {
+        return ZonedDateTime
+                .now(ZoneOffset.UTC).toLocalDateTime();
+    }
+
+    public boolean isSupportedOtpChannel(String channel) {
+        return channel != null && otpChannels.contains(channel.toLowerCase());
+    }
+
+
+
+    public KycAuthResult doKycAuthImpl(String relyingPartyId, String clientId, KycAuthDto kycAuthDto)
```

Comment:
Since we support only the KBA, we can keep this logic in the SunbirdRCAuthenticationService itself and remove this helper service

Reply:
removed